hi-ml/src/health_ml/model_trainer.py

```diff
@@ -215,8 +215,8 @@ def model_train(checkpoint_path: Optional[Path],
     # Create the trainer object. Backup the environment variables before doing that, in case we need to run a second
     # training in the unit tests.
     old_environ = dict(os.environ)
-    # Set random seeds just before training
-    seed_everything(container.get_effective_random_seed())
+    # Set random seeds just before training. Ensure that dataloader workers are also seed correctly.
+    seed_everything(container.get_effective_random_seed(), workers=True)
```

Comment:
See comment above about testing with num_workkers>0

Reply:
see above, I think we want to test that with @mebristo's regression tests rather.